libraries/ui/src/ui/types/ContextAwareProfile.h

```diff
@@ -47,10 +49,18 @@ class ContextAwareProfile : public ContextAwareProfileParent {
     };
 
     ContextAwareProfile(QQmlContext* parent);
+    ~ContextAwareProfile();
+    void onIsRestrictedChanged(bool newValue);
+
     QQmlContext* _context{ nullptr };
-    bool _cachedValue{ false };
-    quint64 _cacheExpiry{ 0 };
-    constexpr static quint64 MAX_CACHE_AGE = MSECS_PER_SECOND;
+    std::atomic<bool> _isRestricted{ false };
+
+private:
+    typedef QSet<ContextAwareProfile*> ContextAwareProfileSet;
+    typedef QMap<QQmlContext*, ContextAwareProfileSet> ContextMap;
```

Comment:
a QMap is actually an ordered map, but we don't care about the order so this could just be a QHash.  a QMap is actually a red-black tree, whereas a QHash is really a hash map